README.md

```diff
@@ -71,6 +71,139 @@ Zerocopy provides byte-order aware integer types that support these
 conversions; see the `byteorder` module. These types are especially useful
 for network parsing.
 
+## Dynamically-sized types
+
+Zerocopy supports slice-based dynamically sized types ("slice DSTs", or just
+"DSTs" for short) via the `KnownLayout` trait.
+
+A slice DST is a type whose trailing field is either a slice or another
+slice DST, rather than a type with fixed size. For example:
+
+```rust
+#[repr(C)]
+struct PacketHeader {
+    ...
+}
+
+#[repr(C)]
+struct Packet {
+    header: PacketHeader,
+    body: [u8],
+}
+```
+
+It can be useful to think of slice DSTs as a generalization of slices - in
+other words, a normal slice is just the special case of a slice DST with
+zero leading fields. In particular:
+- Like slices, slice DSTs can have different lengths at runtime
+- Like slices, slice DSTs cannot be passed by-value, but only by reference
+  or via other indirection such as `Box`
+- Like slices, a reference (or `Box`, or other pointer type) to a slice DST
+  encodes the number of elements in the trailing slice field
+
+### Slice DST layout
+
+Just like other composite Rust types, the layout of a slice DST is not
+well-defined unless it is specified using an explicit `#[repr(...)]`
+attribute such as `#[repr(C)]`. [Other representations are
+supported][reprs], but in this section, we'll use `#[repr(C)]` as our
+example.
+
+A `#[repr(C)]` slice DST is laid out [just like sized `#[repr(C)]`
+types][repr-c-structs], but the presenence of a variable-length field
+introduces the possibility of *dynamic padding*. In particular, it may be
+necessary to add trailing padding *after* the trailing slice field in order
+to satisfy the outer type's alignment, and the amount of padding required
+may be a function of the length of the trailing slice field. This is just a
```

Comment:
"may"? I'm not sure this qualification is technically necessary.

Reply:
Discussed offline; sticking with "may" since there are cases such as this one in which padding is fixed regardless of trailing slice length:

```rust
#[repr(C)]
struct Foo {
    a: u32,
    b: u16,
    z: [[u8; 4]],
}
```